src/Diagnostics.RuntimeHost/Controllers/ArmResource/ArmResourceController.cs

```diff
@@ -21,39 +22,39 @@ public ArmResourceController(IServiceProvider services, IRuntimeContext<ArmResou
         }
 
         [HttpPost(UriElements.Query)]
-        public async Task<IActionResult> ExecuteQuery(string subscriptionId, string resourceGroupName, string provider, string resourceTypeName, string resourceName, [FromBody]CompilationPostBody<dynamic> jsonBody, string startTime = null, string endTime = null, string timeGrain = null, [FromQuery][ModelBinder(typeof(FormModelBinder))] Form Form = null)
+        public Task<IActionResult> ExecuteQuery(string subscriptionId, string resourceGroupName, string provider, string resourceTypeName, string resourceName, [FromBody]CompilationPostBody<dynamic> jsonBody, string startTime = null, string endTime = null, string timeGrain = null, [FromQuery][ModelBinder(typeof(FormModelBinder))] Form Form = null)
         {
-            return await base.ExecuteQuery(new ArmResource(subscriptionId, resourceGroupName, provider, resourceTypeName, resourceName), jsonBody, startTime, endTime, timeGrain, Form: Form);
+            return base.ExecuteQuery(new ArmResource(subscriptionId, resourceGroupName, provider, resourceTypeName, resourceName, GetLocation()), jsonBody, startTime, endTime, timeGrain, Form: Form);
```

Comment:
What will happen if GetLocation() returns null here?

Reply:
location would just be null. detectors need to expect this case happens.